packages/ics721/src/execute.rs

```diff
@@ -297,30 +318,23 @@ where
                 }
                 from_json::<IbcOutgoingProxyMsg>(msg.clone())
                     .ok()
-                    .map(|msg| {
-                        let mut info = info;
-                        match deps.api.addr_validate(&msg.collection) {
-                            Ok(collection_addr) => {
-                                // set collection address as (initial) sender
-                                info.sender = collection_addr;
-                                self.receive_nft(
-                                    deps,
-                                    env,
-                                    info,
-                                    TokenId::new(token_id),
-                                    nft_owner,
-                                    msg.msg,
-                                )
-                            }
-                            Err(err) => Err(ContractError::Std(err)),
-                        }
+                    .map(|msg| match deps.api.addr_validate(&msg.collection) {
+                        Ok(nft_contract) => self.receive_nft(
+                            deps,
+                            env,
+                            &nft_contract,
```

Comment:
...after change, collection is directly passed.

Reply:
yeah much better